### DISCUSSION TITLE
M1 GPU support

### DISCUSSION THREAD

**Discussion by scottleibrand** [openai/whisper#91] [Q&A] [answered]
Has anyone figured out how to make Whisper use the GPU of an M1 Mac?

I can get it to run fine using the CPU (maxing out 8 cores), which transcribes in approximately 1x real time with ----model base.en and ~2x real-time with tiny.en. I'd like to figure out how to get it to use the GPU, but my efforts so far have hit dead ends. Cuda is apparently unsupported on M1 macs, and if I try --device mps I get a warning that `The operator 'aten::repeat_interleave.self_int' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications.` It does indeed: that runs slower than CPU alone.

**Comment by glangford** [comment]
Related
https://github.com/pytorch/pytorch/issues/77764#issuecomment-1254352628

**Comment by elementalcollision** [comment]
Looking forward to this.  Currently running a Medium model against 1.5hrs of audio and the GPU would be a big help. (M1 Max here with 64Gb of Memory)

**Comment by jongwook** [comment, marked as answer]
The PR #382 would be a good place to discuss this; I had a similar issue myself when I tested with a recent nightly version of PyTorch. There might be some combination of versions/settings to make things runnable but I haven't dug deeper.

**Reply by elementalcollision** [reply]
thanks.  I just pulled whisper yesterday but I see that this PR hasn't been committed yet. I will wait to see if it's approved.
